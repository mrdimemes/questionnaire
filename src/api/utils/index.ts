export { default as getFetchError } from "./getFetchError";