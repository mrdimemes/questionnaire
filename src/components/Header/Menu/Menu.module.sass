@use "src/sass/foundation/colors"
@use "src/sass/foundation/breakpoints"

.body
  display: flex
  align-items: center

.interactive
  transition: background-color 0.3s, transform 0.3s
  display: flex
  align-items: center
  column-gap: 45px
  @media (max-width: breakpoints.$tablet)
    position: fixed
    inset: 40px 0 0 50px
    padding: 10px 30px
    flex-direction: column-reverse
    justify-content: flex-end
    align-items: flex-start
    row-gap: 25px
    transform: translateX(100vw)

.theme_dark
  background-color: colors.$dark-theme-2

.theme_light
  background-color: colors.$light-theme-1

.meta
  display: flex
  align-items: center
  column-gap: 30px
  @media (max-width: breakpoints.$tablet)
    width: 100%
    justify-content: space-between

.opened
  transform: translateX(0)
