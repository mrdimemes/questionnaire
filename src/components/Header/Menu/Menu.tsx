import { useState } from "react";
import { Navigation, ThemeSwitchButton, Burger } from "../";
import styles from "./Menu.module.sass";

const Menu = () => {
  const [isOpen, setIsOpen] = useState(false);
  const toggleIsOpen = () => {
    setIsOpen(!isOpen)
  }

  return (
    <div className={styles.body}>
      <Navigation />
      <ThemeSwitchButton />
      <div>user</div>
      <Burger onClick={toggleIsOpen} isOpen={isOpen} />
    </div>
  )
}

export default Menu