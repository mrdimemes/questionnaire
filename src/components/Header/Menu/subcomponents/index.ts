export { Navigation } from "./Navigation";
export { BurgerMenuButton } from "./BurgerMenuButton";
export { ThemeSwitchButton } from "./ThemeSwitchButton";
export { UserPanel } from "./UserPanel";