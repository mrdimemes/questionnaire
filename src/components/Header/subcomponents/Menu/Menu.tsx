import classNames from "classnames";
import { useState } from "react";

import { withTheme } from "src/HOCs";

import {
  Navigation,
  ThemeSwitchButton,
  BurgerMenuButton,
  UserPanel,
} from "./subcomponents";
import styles from "./Menu.module.sass";

import type { MenuProps } from "./types";


const Menu = ({ className }: MenuProps) => {
  const [isOpen, setIsOpen] = useState(false);
  const toggleIsOpen = () => {
    setIsOpen(!isOpen);
  };
  const closeMenu = () => setIsOpen(false);

  return (
    <div className={classNames(styles.Menu, className)}>
      <BurgerMenuButton onClick={toggleIsOpen} isOpen={isOpen} />

      <div className={classNames(
        styles.container,
        isOpen ? styles.opened : null,
      )}>

        <Navigation onClick={closeMenu} />
        <div className={styles.bar}>
          <ThemeSwitchButton />
          <UserPanel />
        </div>

      </div>
    </div >
  );
};

export default withTheme(Menu, styles);