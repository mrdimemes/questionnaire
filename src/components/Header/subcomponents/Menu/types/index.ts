export type { MenuProps } from "./MenuProps";