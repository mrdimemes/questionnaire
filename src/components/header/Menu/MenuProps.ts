export type MenuProps = {
  className?: string,
};