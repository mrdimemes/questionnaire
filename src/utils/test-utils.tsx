import React, { PropsWithChildren } from "react";
import { BrowserRouter } from "react-router-dom";
import { render, RenderOptions } from "@testing-library/react";
import { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore, AppStore, RootState } from "src/redux";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderWithRouters(
  ui: React.ReactElement,
  { ...renderOptions }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <BrowserRouter>{children}</BrowserRouter>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export function renderWithProvidersAndRouters(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}