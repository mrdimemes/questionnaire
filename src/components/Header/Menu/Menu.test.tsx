import { screen } from "@testing-library/react";
import { renderWithProvidersAndRouters } from "src/utils/test-utils";
import Menu from "./Menu";

describe("Header/Menu component", () => {
  test("Menu renders", () => {
    renderWithProvidersAndRouters(<Menu />);
    expect(screen.getByRole("navigation")).toBeInTheDocument();
    expect(screen.getByText(/главная/i)).toBeInTheDocument();
    expect(screen.getByText(/тэги/i)).toBeInTheDocument();
    expect(screen.getByText(/опросы/i)).toBeInTheDocument();
    screen.findAllByRole("button")
      .then((buttons) => expect(buttons).toHaveLength(2))
  });

  test("Menu snapshot", () => {
    const { container } = renderWithProvidersAndRouters(<Menu />);
    expect(container).toMatchSnapshot();
  });
});