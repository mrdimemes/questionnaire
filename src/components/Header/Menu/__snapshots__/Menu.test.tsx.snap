// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header/Menu component Menu snapshot 1`] = `
<div>
  <div
    class="body"
  >
    <div
      class="interactive theme_light"
    >
      <nav
        class="body"
      >
        <a
          class="link"
          href="/"
        >
          Главная
        </a>
        <a
          class="link"
          href="/questionnaires"
        >
          Опросы
        </a>
        <a
          class="link"
          href="/tags"
        >
          Тэги
        </a>
      </nav>
      <div
        class="meta"
      >
        <button
          class="body body"
        >
          <img
            alt="switch theme"
            class="icon"
            src="light-mode-icon.svg"
          />
        </button>
        <div>
          user
        </div>
      </div>
    </div>
    <button
      class="body body"
    >
      <div
        class="icon theme_light"
      />
    </button>
  </div>
</div>
`;
